network-area-diagram/src/test/java/com/powsybl/nad/svg/SubnetworkHighlightTest.java

```diff
@@ -0,0 +1,488 @@
+/**
+ * Copyright (c) 2025, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+package com.powsybl.nad.svg;
+
+import com.google.common.jimfs.Configuration;
+import com.google.common.jimfs.Jimfs;
+import com.powsybl.iidm.network.*;
+import com.powsybl.iidm.network.test.EurostagTutorialExample1Factory;
+import com.powsybl.nad.AbstractTest;
+import com.powsybl.nad.layout.LayoutParameters;
+import com.powsybl.nad.svg.iidm.DefaultLabelProvider;
+import com.powsybl.nad.svg.iidm.NominalVoltageStyleProvider;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.time.ZonedDateTime;
+
+class SubnetworkHighlightTest extends AbstractTest {
+
+    protected java.nio.file.FileSystem fileSystem;
+
+    @BeforeEach
+    void setup() {
+        fileSystem = Jimfs.newFileSystem(Configuration.unix());
+        setLayoutParameters(new LayoutParameters());
+        setSvgParameters(new SvgParameters()
+                .setSvgWidthAndHeightAdded(true)
+                .setFixedWidth(800)
+                .setHighlightGraph(true));
+    }
+
+    @Override
+    protected StyleProvider getStyleProvider(Network network) {
+        return new NominalVoltageStyleProvider(network);
+    }
+
+    @Override
+    protected LabelProvider getLabelProvider(Network network) {
+        return new DefaultLabelProvider(network, getSvgParameters()) {
+        };
+    }
+
+    @Test
+    void testSubnetworkHighlight() {
+        Network network = createWithTieLines();
+        assertSvgEquals("/subnetwork_highlight.svg", network);
```

Comment:
Could you add another test with a partial graph, for instance `VoltageLevelFilter.createVoltageLevelDepthFilter(network, "VLHV2", 1)`?
So that you test the case of a non-visible edge, and so the coverage will be improved.